app/controllers/broadcast_announcements_controller.rb

```diff
@@ -0,0 +1,54 @@
+class BroadcastAnnouncementsController < ApplicationController
+  before_action :set_broadcast_announcement, only: %i[edit update destroy]
+
+  def index
+    @broadcast_announcements = BroadcastAnnouncement.all.select { |announcement| announcement.organization_id == current_organization.id }
```

Comment:
This should be a `where` query rather than `select`, which would load every announcement in the database into memory and then filter them after the fact.